### PR TITLE
Content negotiation

### PR DESCRIPTION
**Pull request by fenekku**
Look at content_negotiation.py diff between latest commit and before

tldr; the code results in same output (I am pretty sure, but we don't have tests yet... ;) ) but the policy (or lack thereof) is much clearer.
 
The problem with content negotiation is that the policy needs to be explicit. 

In the prior code, there was a degenerate case where if the server provided a wildcard, you could get a situation where the best mimetype is a wildcard even though something of same quality with an actual mimetype exists. Potentially worst, is that this result is arbitrary i.e. I don't think the
code was trying to choose one or the other and as such one or the other will be chosen depending on input. So what became apparent is that there isn't a policy for what a "best match is". 

The new code suffers from the same degenerate case, but is explicit about it. And hopefully it is clearer as to what is going on.

That being said we need to establish a policy for "best matchup". What should it be @ppanero @lnielsen  ?

### CODE REVIEW

**Comment by ppanero** [comment]
@fenekku Could you rebase on master? #1 has been merge, and it will greatly help on review since Im quite sure not many files have been changed hehe.

Also we need to merge the black one first, so we avoid more formatting "re-dos".

**Comment by fenekku** [comment]
Closing in favour of #20 -- something weird with this branch not picking up the rebase correctly.
